TST use global_random_seed in sklearn/metrics/tests/test_classification.py

Rishab260 opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

towards #22827
#### What does this implement/fix? Explain your changes.
I added the global_random_seed fixture to the tests:-

- test_matthews_corrcoef_against_numpy_corrcoef
- test_matthews_corrcoef_against_jurman
- test_matthews_corrcoef
- test_matthews_corrcoef_multiclass
- test_matthews_corrcoef_overflow
- test_classification_metric_pos_label_types

#### Any other comments?
This is my first PR ever, feedback and guidance is highly appreciated. Looking forward to contributing to this project more and more in the coming days. Thanks in advance! 

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
https://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

Rishab260 committed:
add global_random_seed to test_matthews_corrcoef_against_numpy_corrcoef

Rishab260 committed:
add global_random_seed to test_matthews_corrcoef_against_jurman

Rishab260 committed:
add global_random_seed to test_matthews_corrcoef

Rishab260 committed:
add global_random_seed to test_matthews_corrcoef_multiclass

Rishab260 committed:
add global_random_seed to test_matthews_corrcoef_overflow

Rishab260 committed:
add global_random_seed to test_classification_metric_pos_label_types

Rishab260 committed:
add global_random_seed to make_prediction

Rishab260 committed:
Undo add global_random_seed to make_prediction

Rishab260 committed:
test_classification [all random seeds]
test_matthews_corrcoef_against_numpy_corrcoef
test_matthews_corrcoef_against_jurman
test_matthews_corrcoef
test_matthews_corrcoef_multiclass
test_classification_metric_pos_label_types

Rishab260 committed:
Merge branch 'scikit-learn:main' into test_classification

Rishab260 committed:
Merge branch 'main' into test_classification

Rishab260 committed:
Merge branch 'main' into test_classification

Rishab260 committed:
Merge branch 'main' into test_classification

yuanx749 commented on sklearn/metrics/tests/test_classification.py:
```diff
@@ -2946,13 +2946,13 @@ def test_balanced_accuracy_score(y_true, y_pred):
     "classes", [(False, True), (0, 1), (0.0, 1.0), ("zero", "one")]
 )
-def test_classification_metric_pos_label_types(metric, classes):
+def test_classification_metric_pos_label_types(metric, classes, global_random_seed):
```
```suggestion
def test_classification_metric_pos_label_types(metric, classes):
```
I think there is no need to apply `global_random_seed` here, since this test only tests if different types of `pos_label` work, which is not a mathematical property.

yuanx749 commented on sklearn/metrics/tests/test_classification.py:
```diff
@@ -2946,13 +2946,13 @@ def test_balanced_accuracy_score(y_true, y_pred):
     No error should be raised for those types.
     """
-    rng = np.random.RandomState(42)
+    rng = np.random.RandomState(global_random_seed)
```
```suggestion
    rng = np.random.RandomState(42)
```

Rishab260 committed:
Update sklearn/metrics/tests/test_classification.py

Co-authored-by: Xiao Yuan <yuanx749@gmail.com>

Rishab260 committed:
Update sklearn/metrics/tests/test_classification.py

Co-authored-by: Xiao Yuan <yuanx749@gmail.com>

Rishab260 commented:
Thanks @yuanx749 for suggesting the changes, changes have been applied, please let me know if this looks good to you now?

yuanx749 reviewed (commented):
LGTM. Let's wait for maintainers to review and approve.

OmarManzoor reviewed (commented):
Thanks for the PR @Rishab260 
Can you please push a commit with [all random seeds] and the tests you updated similar to how it is mentioned in https://github.com/scikit-learn/scikit-learn/issues/28959

Rishab260 commented:
> Thanks for the PR @Rishab260 Can you please push a commit with [all random seeds] and the tests you updated similar to how it is mentioned in #28959

https://github.com/scikit-learn/scikit-learn/pull/30851/commits/282b3af6c4a3cd7ac23be240e896a1eaa3a78a73 is this not it?

OmarManzoor commented:
> Hi @OmarManzoor , [282b3af](https://github.com/scikit-learn/scikit-learn/commit/282b3af6c4a3cd7ac23be240e896a1eaa3a78a73) is this not it? Do I need to do this once more?

I think you may have to push this once more as I can't really see the CI results with that commit.

Rishab260 commented:
> > Hi @OmarManzoor , [282b3af](https://github.com/scikit-learn/scikit-learn/commit/282b3af6c4a3cd7ac23be240e896a1eaa3a78a73) is this not it? Do I need to do this once more?
> 
> I think you may have to push this once more as I can't really see the CI results with that commit.

Okay, doing this again. thanks.

Rishab260 committed:
test_classification [all random seeds]
test_matthews_corrcoef_against_numpy_corrcoef
test_matthews_corrcoef_against_jurman
test_matthews_corrcoef
test_matthews_corrcoef_multiclass
test_matthews_corrcoef_overflow

Rishab260 committed:
Merge branch 'main' into test_classification

OmarManzoor commented:
@Rishab260 The global seeds commit should be the latest one. Since you merged main after that commit the previous CI run gets cancelled.

Rishab260 commented:
> @Rishab260 The global seeds commit should be the latest one. Since you merged main after that commit the previous CI run gets cancelled.

Acknowledged, doing this.

Rishab260 committed:
test_classification [all random seeds]
test_matthews_corrcoef_against_numpy_corrcoef
test_matthews_corrcoef_against_jurman
test_matthews_corrcoef
test_matthews_corrcoef_multiclass
test_matthews_corrcoef_overflow

OmarManzoor reviewed (approved):
LGTM. Thanks @Rishab260
